src/shared/containers/admin/advanced.jsx

```diff
@@ -0,0 +1,286 @@
+    this.setState({ tunnelParams });
+  };
+
+  onSaveBackend() {
+    if (this.state.tunnelParams) {
+      // WIP save tunnel parameters
+      const tunnel = this.state.tunnelParams.active || "None";
+      this.setState({ tunnelParams: null });
+      this.props.apiSetAdminParametersRequest({ tunnel });
+    }
+  }
+
+  displayTunnelDialog() {
+    const { params } = this.props.admin;
+    const backend = this.state.backendParams || params.backend || {};
+    const tunnelParams = this.state.tunnelParams || backend.tunnel || {};
+    const body = (
+      <TunnelBox onChange={this.onChangeTunnel} params={tunnelParams} />
+    );
+    Zrmc.showDialog({
+      header: "Tunnel settings",
+      body,
+      syle: { width: "520px" },
+      onAction: Advanced.onActionTunnel,
+    });
+  }
+
+  onEmailParamsChange = (field, e) => {
+    const { value } = e.currentTarget;
+
+    this.setState((prevState) => ({
+      emailServerParams: {
+        ...prevState.emailServerParams,
+        [field]: value,
+      },
+    }));
+  };
+
+  onSaveEmailParams = () => {
+    const params = this.state.emailServerParams;
+
+    if (
+      params.host.length === 0 ||
+      params.port.length === 0 ||
+      params.username.length === 0 ||
+      params.password.length === 0
+    ) {
+      Zrmc.showDialog({
+        header: "email settings",
+        body: "all parameters are mandatory",
+        syle: { width: "520px" },
+      });
+      return;
+    }
+
+    this.props.apiAdminUpdateRequest({
+      emailServer: params,
+    });
+
+    Zrmc.showDialog({
+      header: "email settings",
+      body: "parameters saved",
+      syle: { width: "520px" },
+    });
+  };
+
+  render() {
+    const emailServer = this.state.emailServerParams;
+    const backend = this.state.backendParams;
+    // const tunnelParams = this.state.tunnelParams || backend.tunnel || {};
+    /* const hasTunnelParams = !!this.state.tunnelParams; */
+    const saveBackendDisabled = !(
+      this.state.backendParams || this.state.tunnelParams
+    );
+    const saveEmailDisabled = !this.state.emailServerParams;
+    return (
+      <Grid>
+        <Inner>
+          <Cell className="mdl-color--white" span={12}>
+            <div style={infoStyleD}>
+              Backend configuration
+              <Button
+                raised
+                disabled={saveBackendDisabled}
+                style={{ float: "right" }}
+                onClick={(e) => {
+                  e.preventDefault();
+                  this.onSaveBackend();
+                }}
+              >
+                SAVE
+              </Button>
+            </div>
+            <form style={infoStyleD}>
+              <div style={{ width: "520px" }}>
+                <TextField
+                  onChange={() => {}}
+                  label="Public Api url"
+                  style={{ width: FORM_WIDTH }}
+                  defaultValue={backend.publicUrl}
+                />
+                <Icon
```

Comment:
The icon is misplaced should be at the end of the textfield:
![screen shot 2018-05-24 at 16 12 03](https://user-images.githubusercontent.com/395718/40490980-3dc21cfa-5f6d-11e8-8417-a174964c157d.png)